src/main/java/com/minecolonies/structures/helpers/StructureProxy.java

```diff
@@ -51,14 +44,19 @@ public StructureProxy(final World worldObj, final String name)
         this.height = size.getY();
         this.length = size.getZ();
 
-        this.blocks = new Block[width][height][length];
-        this.metadata = new IBlockState[width][height][length];
+        this.blocks = new Template.BlockInfo[width][height][length];
+        this.entities = new Template.EntityInfo[width][height][length];
+
+        for(Template.EntityInfo info: structure.getTileEntities())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'info' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)